Subtractions vs negation caused by spaces: `1 - 1` vs `1-1`

### Issue description

A script containing only the line
```java
println( 1 - 1 );
```
works without problem, producing the following line in the log:
`[20:18:34.462][DONE][CLIENT][INFO] 0`
Meanwhile,
```java
println( 1-1 );
```
produces the error
```
[20:20:44.932][DONE][CLIENT][ERROR] Parser Exception @ subtraction_vs_negation.zs:1:10 : ) expected
org.openzen.zenscript.lexer.ParseException: subtraction_vs_negation.zs:1:10: ) expected
	at org.openzen.zenscript.lexer.LLParserTokenStream.required(LLParserTokenStream.java:96)
	at org.openzen.zenscript.parser.expression.ParsedCallArguments.parse(ParsedCallArguments.java:47)
	at org.openzen.zenscript.parser.expression.ParsedExpression.readPostfixExpression(ParsedExpression.java:389)
	at org.openzen.zenscript.parser.expression.ParsedExpression.readUnaryExpression(ParsedExpression.java:349)
	at org.openzen.zenscript.parser.expression.ParsedExpression.readMulExpression(ParsedExpression.java:278)
	at org.openzen.zenscript.parser.expression.ParsedExpression.readAddExpression(ParsedExpression.java:258)
	at org.openzen.zenscript.parser.expression.ParsedExpression.readShiftExpression(ParsedExpression.java:237)
	at org.openzen.zenscript.parser.expression.ParsedExpression.readCompareExpression(ParsedExpression.java:166)
	at org.openzen.zenscript.parser.expression.ParsedExpression.readAndExpression(ParsedExpression.java:156)
	at org.openzen.zenscript.parser.expression.ParsedExpression.readXorExpression(ParsedExpression.java:146)
	at org.openzen.zenscript.parser.expression.ParsedExpression.readOrExpression(ParsedExpression.java:136)
	at org.openzen.zenscript.parser.expression.ParsedExpression.readAndAndExpression(ParsedExpression.java:126)
	at org.openzen.zenscript.parser.expression.ParsedExpression.readOrOrExpression(ParsedExpression.java:110)
	at org.openzen.zenscript.parser.expression.ParsedExpression.readConditionalExpression(ParsedExpression.java:97)
	at org.openzen.zenscript.parser.expression.ParsedExpression.readAssignExpression(ParsedExpression.java:49)
	at org.openzen.zenscript.parser.expression.ParsedExpression.parse(ParsedExpression.java:40)
	at org.openzen.zenscript.parser.statements.ParsedStatement.parse(ParsedStatement.java:299)
	at org.openzen.zenscript.parser.statements.ParsedStatement.parse(ParsedStatement.java:81)
	at org.openzen.zenscript.parser.ParsedFile.parse(ParsedFile.java:215)
	at org.openzen.zenscript.parser.ParsedFile.parse(ParsedFile.java:158)
…
```
I suspect this is an order of operations error. With spaces, it is parsed as `one, subtract, one`, as is expected. Without the spaces, I suspect it's getting parsed as `one, negative one`, and so it's throwing an error because there isn't an operation between the two tokens.
I could be wrong, however, as both
```java
println( 1 * -1 );
```
and
```java
println( 1 * - 1 );
```
produce no errors and print `[20:41:47.319][DONE][CLIENT][INFO] -1` as expected.

### Steps to reproduce

_No response_

### Script used

https://gist.github.com/nerdywhiteguy/ebad0302b0d6eb9736ed3f8fce9a618c#file-subtraction_vs_negation_bugged-zs

### The crafttweaker.log file

https://gist.github.com/nerdywhiteguy/ebad0302b0d6eb9736ed3f8fce9a618c#file-crafttweaker-log

### Minecraft version

1.16

### Forge version

36.1.10

### CraftTweaker version

7.1.0.371

### Other relevant information

There are plenty of other mods and scripts loaded, as I'm in the process of creating a modpack, so you'll see plenty of excess nonsense if you dive into my logs. None of that should effect something so basic as an operator, though.

### The latest.log file

https://gist.github.com/nerdywhiteguy/ebad0302b0d6eb9736ed3f8fce9a618c#latest-log

--- a/Parser/src/main/java/org/openzen/zenscript/parser/expression/ParsedExpression.java
+++ b/Parser/src/main/java/org/openzen/zenscript/parser/expression/ParsedExpression.java
@@ -268,7 +268,17 @@ public abstract class ParsedExpression {
 				ParsedExpression right = readMulExpression(parser.getPosition(), parser, options);
 				left = new ParsedExpressionBinary(position, left, right, OperatorType.CAT);
 			} else {
-				break;
+				// Check if x-1 was scanned as T_INT instead of [T_SUB, T_INT]
+				// If so, replace the Token with the number and treat it as binary call
+				final ZSToken peek = parser.peek();
+				if (peek.content.startsWith("-") && peek.content.length() >= 2) {
+					parser.replace(new ZSToken(peek.type, peek.content.substring(1)));
+					ParsedExpression right = readMulExpression(parser.getPosition(), parser, options);
+					left = new ParsedExpressionBinary(position, left, right, OperatorType.SUB);
+				} else {
+					break;
+				}
+
 			}
 		}
 		return left;

--- a/ScriptingExample/src/test/java/org/openzen/zenscript/scriptingexample/tests/actual_test/arithmethic_operators/AdditionsAndSubtractions.java
+++ b/ScriptingExample/src/test/java/org/openzen/zenscript/scriptingexample/tests/actual_test/arithmethic_operators/AdditionsAndSubtractions.java
@@ -3,9 +3,10 @@ package org.openzen.zenscript.scriptingexample.tests.actual_test.arithmethic_ope
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.CsvSource;
+import org.openzen.zenscript.scriptingexample.tests.helpers.ScriptBuilder;
 import org.openzen.zenscript.scriptingexample.tests.helpers.ZenCodeTest;
 
-public class AdditionTest extends ZenCodeTest {
+public class AdditionsAndSubtractions extends ZenCodeTest {
 
 	@ParameterizedTest(name = "{0} + {1}")
 	@CsvSource({"0,0", "0,1", "10,10", "-1,1", "815,4711"})
@@ -28,5 +29,15 @@ public class AdditionTest extends ZenCodeTest {
 		logger.assertNoWarnings();
 		logger.assertPrintOutputSize(1);
 		logger.assertPrintOutput(0, String.valueOf(55));
+	}
+
+	@Test
+	public void testChainedSubtractions() {
+		ScriptBuilder.create()
+				.add("println(1-2-3-4-5-6-7-8-9-10);")
+				.execute(this);
+
+		logger.assertPrintOutputSize(1);
+		logger.assertPrintOutput(0, String.valueOf(-53));
 	}
 }

